internal/controller/postgrescluster/instance.go

```diff
@@ -1253,6 +1253,16 @@ func (r *Reconciler) reconcileInstance(
 		addDevSHM(&instance.Spec.Template)
 	}
 
+	// mount additional volumes to the Postgres instance containers
+	if err == nil && spec.Volumes != nil && len(spec.Volumes.Additional) > 0 {
+		missingContainers := addAdditionalVolumesToSpecifiedContainers(&instance.Spec.Template, spec.Volumes.Additional)
+
+		if len(missingContainers) > 0 {
+			r.Recorder.Eventf(cluster, corev1.EventTypeWarning, "MissingContainers",
+				"The following containers were specified for additional volumes but are missing: %s.", missingContainers)
```

Comment:
I think what you have is fine, but you expressed dissatisfaction with what you have, so maybe something like this?
```suggestion
			r.Recorder.Eventf(cluster, corev1.EventTypeWarning, "SpecifiedContainerNotFound",
				"The following containers were specified for additional volumes but cannot be found: %s.", missingContainers)
```